docs/api/history.md

```diff
@@ -190,3 +212,133 @@ repo.history.remove(two)
 
 repo.history.toArray() // [root, one, three]
 ```
+
+## Events
+
+History emits events any time something of interest happens. This is
+how Microcosm knows to update the state to accurately reflect what's
+going on given a sequence of actions and their varying statuses.
+
+You can manage event listeners with the following methods.
+
+### `on(event, callback)`
+
+Adds an event listener to a Microcosm History instance.
+```javascript
+const repo = new Microcosm()
+const history = repo.history
+
+history.on('append', callback)
+```
+
+### `off(event, callback)`
+
+Remove an event listener.
+
+```javascript
+// Remove a callback
+history.off('append', callback)
+```
+
+### Event Types
+
+### `append`
+Arguments: `action`
+
+Emitted when an action is pushed onto the History stack.
+
+```javascript
+history.on('append', function(action) {
+  console.log('Action pushed:', action.id)
+})
+
+repo.push(newAction)
+
+// Action pushed: 42
+```
+
+### `remove`
+Arguments: `action`
+
+Emitted when an action is removed from the History stack.
+
+```javascript
+history.on('remove', function(action) {
+  console.log('Action removed:', action.id)
+})
+
+action = repo.push(newAction)
+history.remove(action)
+
+// Action removed: 42
+```
+
+### `update`
+Arguments: `action`
+
+Whenever there is an update to an action's status, an `update` event
+is emitted with that action.
+
+```javascript
+history.on('update', function(action) {
+  console.log('Action status:', action.status)
+})
+
+repo.push(newAction)
+
+// Action status: inactive
+// Action status: open
+// Action status: update
+// Action status: [resolve, reject, cancel]
+```
+
+Whenever an action status updates that precedes other actions in
```

Comment:
thoughts on `Whenever an action that precedes other actions has a status update, ...`?

don't care about the status of the other actions